client/packages/system/src/Encounter/DetailView/DetailView.tsx

```diff
@@ -244,16 +150,19 @@ export const DetailView: FC = () => {
   } = useSaveWithStatusChange(
     saveData,
     data as unknown as EncounterSchema,
-    updateEncounter,
-    nextEncounterModal.toggleOn
+    updateEncounter
   );
 
   const promptToMarkVisitedOnLeaving = useConfirmationModal({
     title: t('heading.are-you-sure'),
     message: t('messages.mark-as-visited'),
     cancelButtonLabel: t('label.leave-as-pending'),
     buttonLabel: t('label.mark-as-visited'),
-    onConfirm: () => saveWithStatusChange(EncounterNodeStatus.Visited),
+    onConfirm: ({ onCancel }) => {
+      saveWithStatusChange(EncounterNodeStatus.Visited, () =>
+        nextEncounterModal.onOpen({ onCancel })
+      );
```

Comment:
Callback here passes through the `onCancel` for prompt-on-leaving... which is to continue the navigation